api/v1beta1/heat_webhook.go

```diff
@@ -94,8 +99,30 @@ func (r *Heat) ValidateCreate() error {
 // ValidateUpdate implements webhook.Validator so a webhook will be registered for the type
 func (r *Heat) ValidateUpdate(old runtime.Object) error {
 	heatlog.Info("validate update", "name", r.Name)
+	oldHeat, ok := old.(*Heat)
+	if !ok {
+		return fmt.Errorf("Expected a Heatv1 object, but got %T", oldHeat)
+	}
+
+	var errors field.ErrorList
+	// We currently have no logic in place to perform database migrations. Changing databases
+	// would render all of the existing stacks unmanageable. We should block changes to the
+	// databaseInstance to protect existing workloads.
+	if r.Spec.DatabaseInstance != oldHeat.Spec.DatabaseInstance {
+		errors = append(errors, field.Invalid(
```

Comment:
should we use `field.Forbidden` instead since we say its a forbidden change?

Reply:
Sounds reasonable, changed this now.